app/Console/Commands/Takedate.php

```diff
@@ -73,13 +73,14 @@ public function handle()
 			$url = Storage::path('big/' . $photo->url);
 			if (file_exists($url)) {
 				$info = $this->metadataExtractor->extract($url, $photo->type);
-				if ($photo->takestamp == '') {
+				if ($info['takestamp'] != null) {
 					$photo->takestamp = $info['takestamp'];
-				}
-				if ($photo->save()) {
-					$this->line($i . ': Takestamp updated for ' . $photo->title);
+					if ($photo->save()) {
+						$this->line($i . ': Takestamp updated for ' . $photo->title);
+					}
+					AlbumUpdate::update_takestamps($photo->album, [$photo->takestamp], true);
```

Comment:
Would this be better before the `}`, so that it only updates if the photo update was successful?